lib/wibox/hierarchy.lua

```diff
@@ -143,12 +143,14 @@ function hierarchy_update(self, context, widget, width, height, region, matrix_t
 
     -- Calculate the draw extents
     local x1, y1, x2, y2 = 0, 0, width, height
-    for _, h in ipairs(self._children) do
-        local px, py, pwidth, pheight = matrix.transform_rectangle(h._matrix, h:get_draw_extents())
-        x1 = math.min(x1, px)
-        y1 = math.min(y1, py)
-        x2 = math.max(x2, px + pwidth)
-        y2 = math.max(y2, py + pheight)
+    if not widget.clip_child_extends then
```

Comment:
If this property has to be canon and used in common widget code base, I'd like it to be documented as such. Can you add it to the common widget doc?

Reply:
I could, but I don't think that would make sense. The API docs document the public, user-facing API. But `clip_child_extends` is internal to the widget system.
It's more akin to `:fit` or `:draw` in that regard.

Either way, I think we should first see what the others think about the functionality/implementation before looking into if/how/where it should be documented.